lib/capify-ec2.rb

```diff
@@ -372,13 +425,55 @@ def deregister_instance_from_elb_by_dns(server_dns)
   end
 
   def reregister_instance_with_target_group_by_dns(server_dns, target_group, timeout)
-    # Stubbed method
-    # Return `reregistered`
+    instance = get_instance_by_dns(server_dns)
+
+    # Check if the instance is a VPC instance, if not return
+    if not is_vpc_instance?(instance)
+        puts "[Capify-EC2] Skipping non-VPC instance '#{server_dns}' ('#{instance}') from target group '#{target_group}'..."
+        return []
+    end
+
+    # Sleep, but why?
+    sleep 10
+    
+    puts "[Capify-EC2] Re-registering instance with ALB target group '#{target_group}'..."
+
+    # Instance registration requires the ALB target group ARN and the instance ID
+    # Obtain target group ARN from target group name assuming the name is unique.
+    target_group_arn = alb_client.describe_target_groups({
+        names: [target_group],
+    })['TargetGroups'][0]['TargetGroupArn']
+
+    # Register the instance with the target group ARN
+    client.register_targets({
+        target_group_arn: target_group_arn,
+        targets: [ { id: instance } ]
+    })
+
+    # Verify instance is attached
+
+    # During registration we would expect to see a state of
+    # initial or healthy.
+    if %w(initial healthy).include?(response.target_health_descriptions[0].target_health.state)
+        puts "[Capify-EC2] Successfully registered '#{server_dns}' ('#{instance}') to target group '#{target_group}'..."
+        reregistered_target_groups << target_group
+    else
+        puts "[Capify-EC2] Failed to add '#{server_dns}' ('#{instance}') to target group '#{target_group}'"
+        puts "[Capify-EC2] Instance is in state '#{response.target_health_descriptions[0].target_health.state}' with description:"
+        puts "[Capify-EC2] #{response.target_health_descriptions[0].target_health.description}"
+    end
+  reregistered_target_groups
```

Comment:
My ruby is a little off... is this important?

Reply:
The last line? It passes a list of all the target groups that had instances removed so it can be used in `lib/capify-ec2/capistrano.rb` to add the instances back in following deployment.